File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/Day.java
package com.tickaroo.tikxml.annotationprocessing.textcontent;

import com.tickaroo.tikxml.annotation.TextContent;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Day {

  @TextContent String name;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Day)) return false;

    Day day = (Day) o;

    return name != null ? name.equals(day.name) : day.name == null;
  }

  @Override public int hashCode() {
    return name != null ? name.hashCode() : 0;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/Day.kt
package com.tickaroo.tikxml.annotationprocessing.textcontent

import com.tickaroo.tikxml.annotation.TextContent
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
class Day {

    @TextContent
    var name: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Day) return false

        val day = other as Day?

        return if (name != null) name == day!!.name else day!!.name == null
    }

    override fun hashCode(): Int {
        return if (name != null) name!!.hashCode() else 0
    }
}
